pkg/ingester/limiter_test.go

```diff
@@ -102,6 +102,96 @@ func TestSeriesLimit_maxSeriesPerMetric(t *testing.T) {
 	}
 }
 
+func TestSeriesLimit_maxMetadataPerMetric(t *testing.T) {
```

Comment:
Initially, I did a complete refactor of the tests to make them accept different types of entities to test (metadata or samples) after the increasing complexity I reverted back and opted to just copy/paste. I'm generally of the opinion that tests should be dumb - I don't expect this to change quick enough to justify making It more complicated for the sake of DRYing up.

Reply:
Different tests are preferable to me too.